--- crypto/pkcs8/p5_pbev2.cc	2026-03-30 19:39:54.519332032 +0000
+++ crypto/pkcs8/p5_pbev2.cc	2026-03-30 19:39:58.465232629 +0000
@@ -63,21 +63,76 @@
      8,
      NID_des_ede3_cbc,
      &EVP_des_ede3_cbc},
+    // 1.3.14.3.2.6
+    {{0x2b, 0x0e, 0x03, 0x02, 0x06},
+     5,
+     NID_des_ecb,
+     &EVP_des_ecb},
+    // 1.3.14.3.2.7
+    {{0x2b, 0x0e, 0x03, 0x02, 0x07},
+     5,
+     NID_des_cbc,
+     &EVP_des_cbc},
+    // 2.16.840.1.101.3.4.1.1
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x01},
+     9,
+     NID_aes_128_ecb,
+     &EVP_aes_128_ecb},
     // 2.16.840.1.101.3.4.1.2
     {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x02},
      9,
      NID_aes_128_cbc,
      &EVP_aes_128_cbc},
+    // 2.16.840.1.101.3.4.1.3
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x03},
+     9,
+     NID_aes_128_ofb128,
+     &EVP_aes_128_ofb},
+    // 2.16.840.1.101.3.4.1.4
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x04},
+     9,
+     NID_aes_128_cfb128,
+     &EVP_aes_128_cfb128},
+    // 2.16.840.1.101.3.4.1.21
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x15},
+     9,
+     NID_aes_192_ecb,
+     &EVP_aes_192_ecb},
     // 2.16.840.1.101.3.4.1.22
     {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x16},
      9,
      NID_aes_192_cbc,
      &EVP_aes_192_cbc},
+    // 2.16.840.1.101.3.4.1.23
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x17},
+     9,
+     NID_aes_192_ofb128,
+     &EVP_aes_192_ofb},
+    // 2.16.840.1.101.3.4.1.24
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x18},
+     9,
+     NID_aes_192_cfb128,
+     &EVP_aes_192_cfb128},
+    // 2.16.840.1.101.3.4.1.41
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x29},
+     9,
+     NID_aes_256_ecb,
+     &EVP_aes_256_ecb},
     // 2.16.840.1.101.3.4.1.42
     {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x2a},
      9,
      NID_aes_256_cbc,
      &EVP_aes_256_cbc},
+    // 2.16.840.1.101.3.4.1.43
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x2b},
+     9,
+     NID_aes_256_ofb128,
+     &EVP_aes_256_ofb},
+    // 2.16.840.1.101.3.4.1.44
+    {{0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x2c},
+     9,
+     NID_aes_256_cfb128,
+     &EVP_aes_256_cfb128},
 };
 
 static const EVP_CIPHER *cbs_to_cipher(const CBS *cbs) {
@@ -276,7 +331,7 @@
   CBS iv;
   if (!CBS_get_asn1(&enc_scheme, &iv, CBS_ASN1_OCTETSTRING) ||
       CBS_len(&enc_scheme) != 0) {
-    OPENSSL_PUT_ERROR(PKCS8, PKCS8_R_UNSUPPORTED_PRF);
+    OPENSSL_PUT_ERROR(PKCS8, PKCS8_R_DECODE_ERROR);
     return 0;
   }
 
