--- a/libarchive/archive_write_add_filter_gzip.c	2025-07-21 06:29:58.505101515 +0000
+++ b/libarchive/archive_write_add_filter_gzip.c	2025-07-21 06:44:09.023676935 +0000
@@ -59,12 +59,13 @@
 	int		 compression_level;
 	int		 timestamp;
 	char	*original_filename;
+	unsigned char	 os;
 #ifdef HAVE_ZLIB_H
 	z_stream	 stream;
 	int64_t		 total_in;
 	unsigned char	*compressed;
 	size_t		 compressed_buffer_size;
-	unsigned long	 crc;
+	uint32_t	 crc;
 #else
 	struct archive_write_program_data *pdata;
 #endif
@@ -108,6 +109,7 @@
 		return (ARCHIVE_FATAL);
 	}
 	f->data = data;
+	data->os = 3; /* default Unix */
 	f->open = &archive_compressor_gzip_open;
 	f->options = &archive_compressor_gzip_options;
 	f->close = &archive_compressor_gzip_close;
@@ -177,6 +179,30 @@
 		return (ARCHIVE_OK);
 	}
 
+	if (strcmp(key, "os") == 0) {
+		if (value == NULL) 
+			return (ARCHIVE_WARN);
+
+		if (strcmp(value, "FAT") == 0) data->os = 0;
+		else if (strcmp(value, "Amiga") == 0) data->os = 1;
+		else if (strcmp(value, "VMS") == 0 || strcmp(value, "OpenVMS") == 0) data->os = 2;
+		else if (strcmp(value, "Unix") == 0) data->os = 3;
+		else if (strcmp(value, "VM") == 0 || strcmp(value, "VM/CMS") == 0) data->os = 4;
+		else if (strcmp(value, "Atari TOS") == 0) data->os = 5;
+		else if (strcmp(value, "HPFS") == 0) data->os = 6; 
+		else if (strcmp(value, "Macintosh") == 0) data->os = 7;
+		else if (strcmp(value, "Z-System") == 0) data->os = 8;
+		else if (strcmp(value, "CP/M") == 0) data->os = 9;
+		else if (strcmp(value, "TOPS-20") == 0) data->os = 10;
+		else if (strcmp(value, "NTFS") == 0) data->os = 11;
+		else if (strcmp(value, "QDOS") == 0) data->os = 12;
+		else if (strcmp(value, "Acorn RISCOS") == 0) data->os = 13;
+		else if (strcmp(value, "Unknown") == 0) data->os = 255;
+		else return (ARCHIVE_WARN);
+
+		return (ARCHIVE_OK);
+	}
+
 	/* Note: The "warn" return is just to inform the options
 	 * supervisor that we didn't handle it.  It will generate
 	 * a suitable error if no one used this option. */
@@ -236,7 +262,7 @@
 	    data->compressed[8] = 4;
     else
 	    data->compressed[8] = 0;
-	data->compressed[9] = 3; /* OS=Unix */
+	data->compressed[9] = data->os;
 	data->stream.next_out += 10;
 	data->stream.avail_out -= 10;
 
