--- a/libarchive/archive_write_add_filter_gzip.c
+++ b/libarchive/archive_write_add_filter_gzip.c
@@ -60,6 +60,7 @@ struct gzip {
 	int		 compression_level;
 	int		 timestamp;
 	char	*original_filename;
+	unsigned char	 os;
 #ifdef HAVE_ZLIB_H
 	z_stream	 stream;
 	uint64_t	 total_in;
@@ -109,6 +110,7 @@ archive_write_add_filter_gzip(struct archive *a)
 	if (gzip == NULL)
 		goto memerr;
 	gzip->original_filename = NULL;
+	gzip->os = 3; /* default Unix */
 #ifdef HAVE_ZLIB_H
 	gzip->compression_level = Z_DEFAULT_COMPRESSION;
 
@@ -184,6 +186,29 @@ archive_compressor_gzip_options(struct archive_write_filter *f, const char *key,
 		}
 		return (ARCHIVE_OK);
 	}
+	if (strcmp(key, "os") == 0) {
+		if (value == NULL)
+			return (ARCHIVE_WARN);
+
+		if (strcmp(value, "FAT") == 0) gzip->os = 0;
+		else if (strcmp(value, "Amiga") == 0) gzip->os = 1;
+		else if (strcmp(value, "VMS") == 0 || strcmp(value, "OpenVMS") == 0) gzip->os = 2;
+		else if (strcmp(value, "Unix") == 0) gzip->os = 3;
+		else if (strcmp(value, "VM") == 0 || strcmp(value, "VM/CMS") == 0) gzip->os = 4;
+		else if (strcmp(value, "Atari TOS") == 0) gzip->os = 5;
+		else if (strcmp(value, "HPFS") == 0) gzip->os = 6;
+		else if (strcmp(value, "Macintosh") == 0) gzip->os = 7;
+		else if (strcmp(value, "Z-System") == 0) gzip->os = 8;
+		else if (strcmp(value, "CP/M") == 0) gzip->os = 9;
+		else if (strcmp(value, "TOPS-20") == 0) gzip->os = 10;
+		else if (strcmp(value, "NTFS") == 0) gzip->os = 11;
+		else if (strcmp(value, "QDOS") == 0) gzip->os = 12;
+		else if (strcmp(value, "Acorn RISCOS") == 0) gzip->os = 13;
+		else if (strcmp(value, "Unknown") == 0) gzip->os = 255;
+		else return (ARCHIVE_WARN);
+
+		return (ARCHIVE_OK);
+	}
 
 	/* Note: The "warn" return is just to inform the options
 	 * supervisor that we didn't handle it.  It will generate
@@ -244,7 +269,7 @@ archive_compressor_gzip_open(struct archive_write_filter *f)
 	} else {
 		gzip->compressed[8] = 0;
 	}
-	gzip->compressed[9] = 3; /* OS=Unix */
+	gzip->compressed[9] = gzip->os;
 	gzip->stream.next_out += 10;
 	gzip->stream.avail_out -= 10;
 
