--- deflate.h
+++ deflate.h
@@ -326,23 +326,7 @@ extern const uint8_t ZLIB_INTERNAL _dist_code[];
     flush = (s->sym_next == s->sym_end); \
   }
 
-#ifdef _MSC_VER
-
-/* MSC doesn't have __builtin_expect.  Just ignore likely/unlikely and
-   hope the compiler optimizes for the best.
-*/
-#define likely(x)       (x)
-#define unlikely(x)     (x)
-
-int __inline __builtin_ctzl(unsigned long mask)
-{
-    unsigned long index ;
-
-    return _BitScanForward(&index, mask) == 0 ? 32 : ((int)index) ;
-}
-#else
 #define likely(x)       __builtin_expect((x),1)
 #define unlikely(x)     __builtin_expect((x),0)
-#endif
 
 #endif /* DEFLATE_H */
