--- a/src/turbojpeg.c
+++ b/src/turbojpeg.c
@@ -1197,12 +1197,14 @@
 #define BITS_IN_JSAMPLE  8
 #include "turbojpeg-mp.c"
 #undef BITS_IN_JSAMPLE
+#ifndef BUN_8BIT_ONLY
 #define BITS_IN_JSAMPLE  12
 #include "turbojpeg-mp.c"
 #undef BITS_IN_JSAMPLE
 #define BITS_IN_JSAMPLE  16
 #include "turbojpeg-mp.c"
 #undef BITS_IN_JSAMPLE
+#endif
 
 /* TurboJPEG 1.2+ */
 DLLEXPORT int tjCompress2(tjhandle handle, const unsigned char *srcBuf,
--- a/src/turbojpeg-mp.c
+++ b/src/turbojpeg-mp.c
@@ -289,6 +289,7 @@
 
 /*************************** Packed-Pixel Image I/O **************************/
 
+#ifndef BUN_8BIT_ONLY
 /* TurboJPEG 3.0+ */
 DLLEXPORT _JSAMPLE *GET_NAME(tj3LoadImage, BITS_IN_JSAMPLE)
   (tjhandle handle, const char *filename, int *width, int align, int *height,
@@ -549,6 +550,7 @@
 
 #endif
 }
+#endif /* BUN_8BIT_ONLY */
 
 
 #undef _JSAMPLE
