diff --git a/deflate.h b/deflate.h
index 9eddb79..ebdd317 100644
--- a/deflate.h
+++ b/deflate.h
@@ -334,12 +334,6 @@ extern const uint8_t ZLIB_INTERNAL _dist_code[];
 #define likely(x)       (x)
 #define unlikely(x)     (x)
 
-int __inline __builtin_ctzl(unsigned long mask)
-{
-    unsigned long index ;
-
-    return _BitScanForward(&index, mask) == 0 ? 32 : ((int)index) ;
-}
 #else
 #define likely(x)       __builtin_expect((x),1)
 #define unlikely(x)     __builtin_expect((x),0)
