--- a/src/hdr_histogram.c
+++ b/src/hdr_histogram.c
@@ -144,7 +144,7 @@ static int64_t counts_get_normalised(const struct hdr_histogram* h, int32_t inde
 static int32_t count_leading_zeros_64(int64_t value)
 {
 #if defined(_MSC_VER) && !(defined(__clang__) && (defined(_M_ARM) || defined(_M_ARM64)))
-    uint32_t leading_zero = 0;
+    unsigned long leading_zero = 0;
 #if defined(_WIN64)
     _BitScanReverse64(&leading_zero, value);
 #else
